RecoMET/METProducers/src/PFMETProducer.cc

```diff
@@ -95,12 +120,50 @@ namespace cms {
 
     //Compute the covariance matrix and fill it
     double sumPtUnclustered = 0;
-    reco::METCovMatrix cov = metSigAlgo_->getCovariance(
-        *inputJets, leptons, candInput, *rho, resPtObj, resPhiObj, resSFObj, event.isRealData(), sumPtUnclustered);
+    reco::METCovMatrix cov;
+    if (applyWeight_)
+      cov = metSigAlgo_->getCovariance(*inputJets,
+                                       leptons,
+                                       candInput,
+                                       *rho,
+                                       resPtObj,
+                                       resPhiObj,
+                                       resSFObj,
+                                       event.isRealData(),
+                                       sumPtUnclustered,
+                                       weights_);
+    else
+      cov = metSigAlgo_->getCovariance(
+          *inputJets, leptons, candInput, *rho, resPtObj, resPhiObj, resSFObj, event.isRealData(), sumPtUnclustered);
 
     return cov;
   }
 
+  void PFMETProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+    edm::ParameterSetDescription desc;
+    desc.add<edm::InputTag>("src", edm::InputTag("particleFlow"));
+    desc.add<bool>("calculateSignificance", false);
+    desc.add<double>("globalThreshold", 0.);
+    desc.add<std::string>("alias", "pfMet");
```

Comment:
this is a rather questionable default, considering how many variants are cloned from this module and that all of them run in the same job in most cases.
```"@module_label"``` seems more appropriate